src/platform/forms-system/src/js/types.js

```diff
@@ -0,0 +1,220 @@
+ * @property {SavedFormMessages} savedFormMessages
+ * @property {SaveInProgress} saveInProgress
+ * @property {boolean} showReviewErrors
+ * @property {(props: any) => JSX.Element} submissionError
+ * @property {(form: any, formConfig: any) => Promise<any>} submit
+ * @property {(props: any) => JSX.Element} submitErrorText
+ * @property {string} submitUrl
+ * @property {string} subTitle
+ * @property {string} title
+ * @property {string} trackingPrefix
+ * @property {(form: any, formConfig: any) => any} transformForSubmit
+ * @property {string} urlPrefix
+ * @property {boolean} useCustomScrollAndFocus
+ * @property {boolean} verifyRequiredPrefill
+ * @property {number} version
+ * @property {string} wizardStorageKey
+ */
+
+/**
+ * @typedef {Object} CustomText
+ * @property {string} appSavedSuccessfullyMessage
+ * @property {string} appType
+ * @property {string} continueAppButtonText
+ * @property {string} reviewPageTitle
+ * @property {string} startNewAppButtonText
+ * @property {string} submitButtonText
+ */
+
+/**
+ * @typedef {Object} SavedFormMessages
+ * @property {string} notFound
+ * @property {string} noAuth
+ */
+
+/**
+ * @typedef {Object} Downtime
+ * @property {Array<string>} dependnecies
+ * @property {string} endTime
+ * @property {string} message
+ * @property {boolean} requiredForPrefill
+ * @property {'down' |'downtimeApproaching' | 'ok'} status
+ * @property {string} startTime
+ */
+
+/**
+ * @typedef {Object} PreSubmitInfo
+ * @property {(props: any) => JSX.Element} CustomComponent
+ * @property {string} error
+ * @property {string} field
+ * @property {JSX.Element} label
+ * @property {string | JSX.Element} notice
+ * @property {boolean} required
+ */
+
+/**
+ * @typedef {Object} SaveInProgress
+ * @property {Object} messages
+ * @property {string} messages.inProgress
+ * @property {string} messages.expired
+ * @property {string} messages.saved
+ */
+
+/**
+ * @typedef {Object} FormConfigChapter
+ * @property {Record<string, FormConfigPage>} pages
+ * @property {string | Function} title
+ */
+
+/**
+ * @typedef {Object} FormConfigPage
+ * @property {string} arrayPath
+ * @property {(props: any) => JSX.Element} CustomPage
+ * @property {(props: any) => JSX.Element} CustomPageReview
+ * @property {(formData: Object) => boolean} depends
+ * @property {Object} initialData
+ * @property {(formData: any) => void} onContinue
+ * @property {(data: any) => boolean} itemFilter
+ * @property {string} path
+ * @property {SchemaOptions} schema
+ * @property {string | Function} scrollAndFocusTarget
+ * @property {boolean} showPagePerItem
+ * @property {string | Function} title
+ * @property {UISchemaOptions} uiSchema
+ * @property {(item, index) => void} updateFormData
+ */
+
+/**
+ * @typedef {Object} PageSchema - The schema for a page (only uiSchema and schema). If you want all the page properties, use FormConfigPage
+ * @property {UISchemaOptions} uiSchema
+ * @property {SchemaOptions} schema
+ */
+
+/**
+ * @typedef {{
```

Comment:
This syntax needs to be used when we have `[key:string]: Type`, or if you have colons in the key names . However this syntax does not allow for comments.